### PR TITLE
Cypress page access tests by pg role

### PR DESCRIPTION
**Pull request by dleard**
- Adds spec tests for each role asserting that all pages that role should be able to access, can be accessed.
 - Set xo to ignore the cypress directory

### CODE REVIEW

File: app/cypress/integration/admin-access-all-pages.spec.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+describe('When logged in as an analyst', () => {
+  before(() => {
```
Comment (matthieu-foucault) [inline app/cypress/integration/admin-access-all-pages.spec.js]:
```suggestion
  beforeEach(() => {
```

File: app/cypress/integration/admin-access-all-pages.spec.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    cy.login(
+      Cypress.env('TEST_ADMIN_USERNAME'),
+      Cypress.env('TEST_ADMIN_PASSWORD')
+    );
+  });
+
+  after(() => {
```
Comment (matthieu-foucault) [inline app/cypress/integration/admin-access-all-pages.spec.js]:
```suggestion
  afterEach(() => {
```

File: app/cypress/integration/analyst-access-all-pages.spec.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+describe('When logged in as an analyst', () => {
+  before(() => {
```
Comment (matthieu-foucault) [inline app/cypress/integration/analyst-access-all-pages.spec.js]:
```suggestion
  beforeEach(() => {
```

File: app/cypress/integration/analyst-access-all-pages.spec.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    cy.login(
+      Cypress.env('TEST_ANALYST_USERNAME'),
+      Cypress.env('TEST_ANALYST_PASSWORD')
+    );
+  });
+
+  after(() => {
```
Comment (matthieu-foucault) [inline app/cypress/integration/analyst-access-all-pages.spec.js]:
```suggestion
  afterEach(() => {
```

File: app/cypress/integration/reporter-access-all-pages.spec.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
+describe('When logged in as a reporter', () => {
+  before(() => {
```
Comment (matthieu-foucault) [inline app/cypress/integration/reporter-access-all-pages.spec.js]:
```suggestion
  beforeEach(() => {
```

File: app/cypress/integration/reporter-access-all-pages.spec.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    cy.logout();
+    cy.sqlFixture('reporter-all-access-setup');
+    cy.login(
+      Cypress.env('TEST_REPORTER_USERNAME'),
+      Cypress.env('TEST_REPORTER_PASSWORD')
+    );
+  });
+
+  after(() => {
```
Comment (matthieu-foucault) [inline app/cypress/integration/reporter-access-all-pages.spec.js]:
```suggestion
  afterEach(() => {
```

File: app/package.json [hunk 7]
```diff
@@ -23,6 +23,9 @@
     "storybook": "start-storybook -s ."
   },
   "xo": {
+    "ignores": [
+      "cypress"
+    ],
```
Comment (matthieu-foucault) [inline app/package.json]: That's a bit excessive. We can override the xo config and disable the one rule that should be disabled

Comment (matthieu-foucault) [reply app/package.json]: See https://github.com/xojs/xo#config-overrides

**Comment by dleard** [comment]
> * Adds spec tests for each role asserting that all pages that role should be able to access, can be accessed.
> * Set xo to ignore the cypress directory for rules:
  - "jest/valid-expect-in-promise",
  - "promise/prefer-await-to-then"

**Pull request merged**